### PR TITLE
cqed

### PR DESCRIPTION
**Pull request by amirhassank**
# Changes made

# Justification

### CODE REVIEW

File: content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb [hunk 1]
```diff
@@ -0,0 +1,764 @@
...
+    "plt.show()\n",
+    "\n",
+    "print(r'gamma=  {:.2f} kHz'.format(g/1e3))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### In this chapter we introduce the Jaynes-Cummings model and we consider the relevant quantities for a system consisting of a qubit coupled to a cavity. We use Qiskit Pulse to extract the qubit-resonator coupling strength $g$, the qubit spontaneous emission rate $\\gamma$, and the cavity decay rate $\\kappa$. These parameters combined with the qubit frequency and the resonator frequency that we measured in a previous <a href=\"https://qiskit.org/textbook/ch-quantum-hardware/cQED-JC-SW.html\">chapter</a> allow us to describe our qubit-resonator system."
```
Comment (frankharkins) [inline content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb:713]: Looks like you've left a note in here? Maybe this should be in the introduction.

File: content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb [hunk 2]
```diff
@@ -0,0 +1,764 @@
...
+    "\n",
+    "In the limit that detuning between the qubit and the resonator $\\Delta=\\omega_q-\\omega_r$ is less than the coupling strength between the two, $|\\Delta|\\ll g$, the resonator-qubit system becomes hybridized, leading to coherent excitation swaps which can be useful for certain two-qubit operations. However, for optimal readout, we want to operate the system in the dispersive limit, where the qubit-resonator detuning is much larger than the coupling rate and the resonator decay rate: $|\\Delta| \\gg g,\\kappa$. In this limit the interaction between the qubit and resonator influences each of their frequencies, a feature that can be used for measuring the state of the qubit. We can apply the dispersive approximation in the limit of few photons in the resonator, and approximate the JC Hamiltonian using second-order perturbation theory as: \n",
+    "\n",
+    "$H_{JC(disp)}=(\\omega_r+ \\chi \\sigma_z) a^\\dagger a + \\frac{1}{2} \\tilde{\\omega}_q \\sigma_z$\n",
```
Comment (frankharkins) [inline content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb:33]: Is there a hbar missing here?

File: content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb [hunk 3]
```diff
@@ -0,0 +1,764 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "from qiskit import pulse            # This is where we access all of our Pulse features!\n",
+    "from qiskit.pulse import Play, Acquire\n",
+    "from qiskit.pulse import pulse_lib\n",
+    "import numpy as np\n",
+    "\n",
+    "dt=backend_config.dt   # AWG resolution\n",
```
Comment (frankharkins) [inline content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb:182]: What does AWG mean? I don't think it's explained in this chapter

Comment (diemilio) [reply content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb:182]: Arbitrary waveform generator

Comment (frankharkins) [reply content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb:182]: If it's not mentioned in the chapter, I think it should be included

File: content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb [hunk 4]
```diff
@@ -0,0 +1,764 @@
...
+    "# in a reflective measurement the data is encoded in the phase of the output signal\n",
+    "def process_reflective_measurement(freqs,values):\n",
+    "    phase_grad = np.gradient(savgol_filter(np.unwrap(np.angle(values)),3,2),freqs)\n",
+    "    return (phase_grad-min(phase_grad))/(max(phase_grad)-min(phase_grad)) - 1\n",
+    "\n",
+    "# lorentzian function\n",
+    "def lorentzian(f, f0, k, a, offs):\n",
+    "    return -a*k/(2*np.pi)/((k/2)**2+(f-f0)**2)+offs\n",
+    "\n",
+    "# fit a lorentizan function\n",
```
Comment (frankharkins) [inline content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb:131]: This comment is not especially helpful as the reader could probably guess this from the function name. I would go for a description of its behaviour, e.g. 
>`fit_lorentzian` takes two arrays that contain the x & y values of each data point respectively. It returns an estimate for the parameters that best reproduce this data. Output is... (explain what `popt` and `pcov` are).

File: content/ch-quantum-hardware/ac-Stark-shift.ipynb [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+    "plt.xlabel(r'Resonator tone amplitude [V]')\n",
+    "plt.ylabel(r'FWHM (MHz)')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### In this chapter we discuss the ac-Stark shift that the qubit experiences due to the presence of photons in the resonator. We use Qiskit Pulse to measure the qubit frequency shift and linewidth broadening"
```
Comment (frankharkins) [inline content/ch-quantum-hardware/ac-Stark-shift.ipynb:414]: Same here, I think this is meant for the introduction

File: content/ch-quantum-hardware/ac-Stark-shift.ipynb [hunk 6]
```diff
@@ -0,0 +1,465 @@
...
+    "from qiskit.pulse import pulse_lib\n",
+    "import numpy as np\n",
+    "\n",
+    "qubit=0   # qubit used in our experiment\n",
+    "\n",
+    "inst_sched_map = backend_defaults.instruction_schedule_map\n",
+    "# Get the default measurement pulse sequence\n",
+    "measure = inst_sched_map.get('measure', qubits=backend_config.meas_map[0])  \n",
+    "\n",
+    "qubit_drive_sigma = 100e-9\n",
```
Comment (frankharkins) [inline content/ch-quantum-hardware/ac-Stark-shift.ipynb:156]: Can you explain where these values come from please?

File: content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb (hunk 1 above)
Comment (amirhassank) [reply content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb:713]: We do also mention this in the introduction. It'd be nice to keep this text in the conclusion

File: content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb (hunk 3 above)
Comment (amirhassank) [reply content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb:182]: That's a rabbit hole that I don't theink we should climb down in this chapter ... I replace the term AWG with Hardware

File: content/ch-quantum-hardware/ac-Stark-shift.ipynb (hunk 5 above)
Comment (amirhassank) [reply content/ch-quantum-hardware/ac-Stark-shift.ipynb:414]: We do also mention this in the introduction. It'd be nice to keep this text in the conclusion

File: content/ch-quantum-hardware/ac-Stark-shift.ipynb (hunk 6 above)
Comment (amirhassank) [reply content/ch-quantum-hardware/ac-Stark-shift.ipynb:156]: The choice was a bit arbitrary, but I'll write a comment

Comment (frankharkins) [reply content/ch-quantum-hardware/ac-Stark-shift.ipynb:156]: Thank you

**Review by frankharkins — APPROVED** [review]

**Pull request merged**